Subscription view for managing feeds and Navigation drawer feed list …

raghulj opened the pull request:
A new UI for subscription view support in navigation drawer and feed subscriptions.  More information about these changes can be found in #836

TomHennen commented:
I do really like the way this looks. It's probably the way to go. I'm still not sure if we want to get rid of the subscriptions from the drawer.  Since we're thinking about making other layout and visual changes in 1.3 I'm going to wait on merging this until we have a coherent strategy.

mfietz commented on app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java:
```diff
@@ -537,7 +550,7 @@ protected void onPostExecute(DBReader.NavDrawerData result) {
                 if(!ArrayUtils.contains(NAV_DRAWER_TAGS, lastFragment)) {
                     long feedId = Long.valueOf(lastFragment);
-                    loadFeedFragmentById(feedId);
+                    //loadFeedFragmentById(feedId);
```
Hm, why is this commented out?

raghulj replied:
The idea is to refactor the whole junk as we are not showing the feed list in the navigation bar.  Since, a lot of changes are involved, I am reverting back the code.  But, I believe, In future we could just refresh the count of "Queue" and "New" instead of the whole feed list.

mfietz commented:
Also, this cannot be merged at the moment, please rebase on current develop branch

mfietz commented on app/src/main/java/de/danoeh/antennapod/adapter/NavListAdapter.java:
```diff
@@ -119,7 +123,7 @@ private Drawable getDrawable(String tag) {
     @Override
     public int getCount() {
-        return getSubscriptionOffset() + itemAccess.getCount();
+        return getSubscriptionOffset() ;//+ itemAccess.getCount(); //Avoid feed
```
I believe this peace of code is the reason I do not see any feeds in the drawer anymore

raghulj replied:
Yes.

mfietz commented on app/src/main/java/de/danoeh/antennapod/fragment/SubscriptionFragment.java:
```diff
@@ -0,0 +1,108 @@
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
```
You should register with the EventDistributor and react to EventDistributor.FEED_LIST_UPDATE (feeds might be downloaded in a background thread and you probably want to display them as soon as they are processed)

raghulj replied:
I have added the code to refresh the feed based on the listener in the new commit. Please check.

mfietz commented:
I'm really not sure about displaying the time of last update. Honestly, doesn't really look very useful to me (usually, the "X min ago" will be the same for all feeds). As of now, we do not use this information anywhere - I'm not sure users will understand what it means immediately, could bring a lot more confusion than clarity.

It would make a lot more sense to me to show when the most current episode of a podcast was published. But I would be able to live without that information, too...

raghulj commented:
@mfietz  I do agree about the "X min ago". I have removed that for now. I was thinking about showing the latest updated time of podcast and not the downloading time.  I have rebased on develop. Could you please let me know if I need to send a pull request?

mfietz commented on app/src/main/java/de/danoeh/antennapod/fragment/SubscriptionFragment.java:
```diff
@@ -0,0 +1,128 @@
+    }
+
+    private void refreshSubscriptionList() {
+        Observable.just(loadData()).subscribe(new Action1<DBReader.NavDrawerData>() {
```
Just for documentation: This will still run on the main thread!
Replace with 
```
Observable.defer(() -> Observable.just(loadData()))
                .subscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(...);
```

keunes commented:
I think people responded really positively to this idea, with just some implementation details keeping away from moving forward. @TomHennen, would it be an to try and attach a concrete milestone to this?

TomHennen commented:
There's probably space now that new and all episodes are collapsed. 1.4
could work.

On Wed, Sep 9, 2015, 7:17 PM keunes <notifications@github.com> wrote:

> I think people responded really positively to this idea, with just some
> implementation details keeping away from moving forward. @TomHennen
> <https://github.com/TomHennen>, would it be an to try and attach a
> concrete milestone to this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AntennaPod/AntennaPod/pull/848#issuecomment-139069975>
> .
>

TomHennen commented:
Please consider resubmitting against the latest in develop. I think there's room for a dedicated subscriptions option in the nav drawer now.
